design/recovery-internals.md

```diff
@@ -0,0 +1,142 @@
+# FDB Recovery Internals
+
+FDB uses recovery to handle various failures, such as hardware and network failures. When the current transaction system no longer works properly due to failures, recovery is automatically triggered to create a new generation of the transaction system. 
+
```

Comment:
Can we be a bit more specific about failure scenarios? Give examples of some of hardware failures? Can a process death cause recovery? Can StorageServer death cause recovery?

Reply:
I have added the `Background` section in the doc to answer these questions.  :)